fastly/cloudfiles.go

```diff
@@ -0,0 +1,255 @@
+	resp, err := c.Get(path, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	var cloudfiles *Cloudfiles
+	if err := decodeJSON(&cloudfiles, resp.Body); err != nil {
+		return nil, err
+	}
+	return cloudfiles, nil
+}
+
+// UpdateCloudfilesInput is used as input to the UpdateCloudfiles function.
+type UpdateCloudfilesInput struct {
+	// Service is the ID of the service. Version is the specific configuration
+	// version. Both fields are required.
+	Service string
+	Version int
+
+	// Name is the name of the Cloudfiles to update.
+	Name string
+
+	NewName           string `form:"name,omitempty"`
+	AccessKey         string `form:"access_key,omitempty"`
+	BucketName        string `form:"bucket_name,omitempty"`
+	Path              string `form:"path,omitempty"`
+	Region            string `form:"region,omitempty"`
+	Placement         string `form:"placement,omitempty"`
+	Period            uint   `form:"period,omitempty"`
+	GzipLevel         uint   `form:"gzip_level,omitempty"`
+	Format            string `form:"format,omitempty"`
+	FormatVersion     uint   `form:"format_version,omitempty"`
+	ResponseCondition string `form:"response_condition,omitempty"`
+	MessageType       string `form:"message_type,omitempty"`
+	TimestampFormat   string `form:"timestamp_format,omitempty"`
+	PublicKey         string `form:"public_key,omitempty"`
+}
+
+// UpdateCloudfiles updates a specific Cloudfiles.
+func (c *Client) UpdateCloudfiles(i *UpdateCloudfilesInput) (*Cloudfiles, error) {
+	if i.Service == "" {
+		return nil, ErrMissingService
+	}
+
+	if i.Version == 0 {
+		return nil, ErrMissingVersion
+	}
+
+	if i.Name == "" {
+		return nil, ErrMissingName
+	}
+
+	path := fmt.Sprintf("/service/%s/version/%d/logging/cloudfiles/%s", i.Service, i.Version, url.PathEscape(i.Name))
+	resp, err := c.PutForm(path, i, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	var cloudfiles *Cloudfiles
+	if err := decodeJSON(&cloudfiles, resp.Body); err != nil {
+		return nil, err
+	}
+	return cloudfiles, nil
+}
+
+// DeleteCloudfilesInput is the input parameter to DeleteCloudfiles.
+type DeleteCloudfilesInput struct {
+	// Service is the ID of the service. Version is the specific configuration
+	// version. Both fields are required.
+	Service string
+	Version int
+
+	// Name is the name of the Cloudfiles to delete (required).
+	Name string
+}
+
+// DeleteCloudfiles deletes the given Cloudfiles version.
+func (c *Client) DeleteCloudfiles(i *DeleteCloudfilesInput) error {
+	if i.Service == "" {
+		return ErrMissingService
+	}
+
+	if i.Version == 0 {
+		return ErrMissingVersion
+	}
+
+	if i.Name == "" {
+		return ErrMissingName
+	}
+
+	path := fmt.Sprintf("/service/%s/version/%d/logging/cloudfiles/%s", i.Service, i.Version, url.PathEscape(i.Name))
+	resp, err := c.Delete(path, nil)
+	if err != nil {
+		return err
+	}
+
+	var r *statusResp
+	if err := decodeJSON(&r, resp.Body); err != nil {
+		return err
+	}
+	if !r.Ok() {
+		return fmt.Errorf("Not Ok")
```

Comment:
I can see we are using this in all of your other PRs too, maybe we should consider moving to an errors constant in [`errors.go`](https://github.com/fastly/go-fastly/blob/master/fastly/errors.go) and make the message more descriptive.

Reply:
I've addressed this in https://github.com/fastly/go-fastly/pull/162/commits/8fc2b9cbc39c3e0906b61233959df938d6d3dcfb.